lib/media.js

```diff
@@ -700,12 +701,62 @@ Cam.prototype.getSnapshotUri = function(options, callback) {
 		'</GetSnapshotUri>' +
 		this._envelopeFooter()
 	}, function(err, data, xml) {
+		if (!err) {
+			this.snapshotUri = linerase(data).getSnapshotUriResponse.mediaUri;
+		}
 		if (callback) {
-			callback.call(this, err, err ? null : linerase(data).getSnapshotUriResponse.mediaUri, xml);
+			callback.call(this, err, this.snapshotUri, xml);
 		}
 	}.bind(this));
 };
 
+/**
+ * Receive snapshot
+ * For this method to work, you must have previously called the #getSnapshotUri method.
+ * @returns Promise - on success { mimeType:string, prefix:string, rawImage:string }
+ */
+Cam.prototype.getSnapshot = function() {
+	const self = this;
+	return new Promise(function(resolve, reject) {
+		if (!self.snapshotUri) {
```

Comment:
Two things here:

1. I don't like the idea of having the dependancy to have to call getSnapshotUri first - this call should automatically do that don't you think? We can just pass the options through
2. Based on 1, we should potentially cache somewhere Uri's for the different `profileToken`s, because calling this on an encoder should enable us to get every input's video (as well as different res images from the different profiles of a single camera)

Thoughts on the above?

Reply:
1. I currently get around 5 snapshots per second in the code I am working on. I don't think it'd be a good idea to call getSnapshotUri() each time. It's better to cache it. Or, make the User cache it and pass it in.
2. I'm not working with encoders at this time, so my experience there is relatively minimal. I just wouldn't know. :(

Reply:
No I agree it should be cached for a period of time as it will definitely make calls faster, I just think that it should happen automatically.

Make getSnapshot have an optional `profileToken` argument, that is passed to getSnapshotUri, otherwise the default is used. Keep a cache of the snapshot uri's in memory based on the different `profileTokens`.

I can try and draft up an example?